### DISCUSSION TITLE
Environment variable not resolve in my script

### DISCUSSION THREAD

**Discussion by eaxdev** [community/community#26230] [Actions]
<p>Hello. I use Github Actions and and faced a problem. I defined env variable like this:</p>
<pre><code>-name: Publish with Gradle
env:
RELEASE\_VERSION: ${GITHUB_REF:11}
uses: eskatos/gradle-command-action@v1
with:
gradle-version: current
arguments: test publish 
</code></pre>
<p>Than, I want to use my RELEASE_VERSION in my gradle script.</p>
<p>I try get it:</p>
<pre><code>version = System.getenv('RELEASE\_VERSION')  
</code></pre>
<p>but instead of getting the name of my tag, I get this variable as is, i.e.</p>
<pre><code>version = ${GITHUB_REF:11}
</code></pre>
<p> Why was the value not calculated and assigned to a variable? I expected to get:</p>
<pre><code>version = 1.0.0
</code></pre>
<p>Job run on ubuntu-latest.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>When you set value for an nvironment variable, it doesn’t support geting substring of another env. You need to run it in bash script and use set-env to create env: RELEASE_VERSION. </p>
<p>Please see my example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181096160-2d7bed38-519b-4498-bff5-4d51cf25aff9.png" alt="set-env.png" data-base62-sha1="zIiKTAfUsendG3MRC1lCSEroB2F" width="435" height="240"></p>
<p>Note: Is your GITHUB_REF like refs/tags/1.0.0 or refs/heads/1.0.0? If it is the second one, then ${GITHUB_REF:11} is right. </p>
